src/dbally/audit/event_handlers/base.py

```diff
@@ -26,13 +26,13 @@ async def request_start(self, user_request: RequestStart) -> RequestCtx:
         """
 
     @abc.abstractmethod
-    async def event_start(self, event: Union[LLMEvent], request_context: RequestCtx) -> EventCtx:
+    async def event_start(self, event: Union[LLMEvent, SimilarityEvent], request_context: RequestCtx) -> EventCtx:
```

Comment:
I'm not a fan of those Unions here, in separate PR I will do a refactor of audit module and create a base class for events.